src/base/msg_buffer.rs

```diff
@@ -0,0 +1,21 @@
+macro_rules! buffer_of_size {
+    ($size: expr) => {
+        [0u8; $size]
+    };
+}
+pub(crate) use buffer_of_size;
+
+macro_rules! log_to {
+    ($logger: expr, $base: expr) => {
+        $logger.log($base);
+    };
+}
```

Comment:
Again, why do we need a macro for this? Isn't `usb_logger.log("foo")` good?

Reply:
It is there more for consistency, than anything :) - So the user doesn't need to remember "Ah! I don't need to use `strfmt` here, therefore I should use an entirely different syntax to do this action!"

It'd be like if you had `fmt.printf("Abc {}", 123)` and `system.do_log("Abc 123")` - the fact that when you want to print something you have to think "Am I formatting something" and then **figure out** which object you want to query, is needless use of a human's working memory

Anywho, all that being said - I'm kind of indifferent to it 🤷. If you think using a macro in one situation and using a function in another is easy enough to remember, we can remove this!

Reply:
Oh maybe I didn't play with this enough... but the underlying `log` call isn't a macro, so I assumed there was no fancy pattern matching going on here, like there is with `println!`. If that was the case I would have expected to see `$base` passed as a `tt` and having the cases fall through to the underlying `log!` macro.